backend/dataall/core/stacks/api/resolvers.py

```diff
@@ -57,6 +58,13 @@ def resolve_events(context, source: Stack, **kwargs):
     return json.dumps(source.events or {})
 
 
```

Comment:
We can take inspiration from the resolver in shares_base/api and add guardrails for exceptions + move all logic to service.

shares_base resolver:
```
def resolve_can_view_logs(context: Context, source: ShareObject):
    try:
        return ShareLogsService.check_view_log_permissions(context.username, context.groups, source.shareUri)
    except Exception as e:
        log.error(f'Failed to resolve if user can view share logs due to: {e}')
        return False
```
equivalent for resolve_stack_visibility:
```
```
def resolve_can_view_logs(context: Context, source: Stack, **kwargs):
   if not source:
        return False
    try:
        return StackService.check_if_user_allowed_view_logs(context.username, context.groups, source.stackUri)
    except Exception as e:
        log.error(f'Failed to resolve if user can view logs due to: {e}')
        return False
```

Reply:
Coming back to the example of shares_base/resolver ---> the resolver does not contain any business logic, it just calls the service. The service `check_if_user_allowed_view_logs` is then the one decorated and contains the permission logic. The service is used by the resolver and also by other services such as `get_stack_logs` which does not need to be decorated

Reply:
class StackService:
    @is_feature_has_allowed_values(
        allowed_values=['admin-only', 'enabled', 'disabled'],
        enabled_values=['admin-only', 'enabled'],
        default_value='enabled',
        resolve_property=StackServiceUtils.map_target_to_config,
    )
    @staticmethod
    def check_if_user_allowed_view_logs(groups, target_type):
        config = config.get_property(StackServiceUtils.map_target_to_config(target_type), 'enabled')
        if config == 'admin-only' and 'DAAdministrators' not in groups:
            return False
        return True

Reply:
I think I misunderstood you in your previous comment. I thought you mean to replicate logic I had `resolve_can_view_logs` to the other resolver. Now I can get what you mean. 
No exceptions ( guard rails ) , no business logic but only validations in the resolver layer.

Reply:
I saw this comment - https://github.com/data-dot-all/dataall/pull/1559#discussion_r1766409253 and thought you like the exceptions in the resolver and thus implemented this way. 
I will make changes

Reply:
Almost almost :) Let's do the analysis again:

In the resolver that resolves the stack/share_logs visibility (canViewLogs value):
- call the StackLogsService or the ShareLogsService `check_view_logs_permissions`
- add try/except clause to ensure it defaults to returning False in case there in an exception - otherwise the whole ShareObject/Environment/Dataset won't be able to be returned

The `check_view_logs_permissions`:
- should be decorated with `is_feature_enabled_for_allowed_values` - raises Exceptions for disabled config values
- inside it should contain additional logic for `admin-only` and other ownership checks - returns False when not alowed - ⚠️ We should change this behavior and raise an Exception as we end up doing in `get_share_logs` for example:
```
exceptions.ResourceUnauthorized(
                username=context.username,
                action='View Share Logs',
                resource_uri=shareUri,
            )
```

Any other resolver that uses `check_view_logs` to return logs (e.g. `get_share_logs`):
- directly use check_view_logs - which should return the exception

Reply:
Thanks once again @dlpzx , for the review and the explanation. I have made the updates. Please let me know if they look good